src/rules/no-snapshots.js

```diff
@@ -0,0 +1,37 @@
+import { getDocsUrl } from './util';
+
+export default {
+  meta: {
+    docs: {
+      description: 'Disallow snapshots',
+      uri: getDocsUrl(__filename),
+    },
+    messages: {
+      noSnapshots:
+        'Do not use {{method}} or related methods that generate snapshots.',
+    },
+  },
+
+  create(context) {
+    return {
+      CallExpression({ callee: { property } }) {
+        if (isSnapshot(property && property.name)) {
```

Comment:
It'd be better to do the existence check outside of the method call, since that way you'd only do the check if the `property` is truthy.